rsync_to_rdisc.py

```diff
@@ -69,124 +64,201 @@ def send_email(sender, receivers, subject, text, attachment=None):
+                "[[ -f {0}{1}/{2} ]] && echo \"Present\" "
+                "|| echo \"Absent\""
+            ).format(folder["input"], run, check_file))
+            status = stdout.read().decode("utf8").rstrip()
+            if status == "Absent":
+                missing.append(check_file)
+    return missing
+
+
+def action_if_file_missing(folder, continue_rsync, remove_run_file, missing, run):
+    if 'continue_without_email' in folder and folder["continue_without_email"]:
+        # Do not send a mail
+        pass
+    elif 'continue_without_email' in folder and not folder["continue_without_email"]:
+        # Send a mail and lock datatransfer
+        reason = (
+            "Analysis not complete (file(s) {0} missing). "
+            "Run = {1} in folder {2} ".format(" and ".join(missing), run, to_be_transferred[run]))
+        make_mail(run, "notcomplete", reason, run_file)
+        continue_rsync = False
+        remove_run_file = False
+    else:  # Send a mail and lock datatransfer
+        reason = ("Unknown status {0} in settings.py for {1}").format(folder["continue_without_email"], folder)
+        make_mail(run, "settings", reason, run_file)
+        continue_rsync = False
+        remove_run_file = False
+
+    return continue_rsync, remove_run_file
+
+
+def rsync_server_remote(settings, hpc_server, client, to_be_transferred):
+    date = str(datetime.datetime.now()).split(".")[0]
+    bgarray_log_file = "{bgarray}/{log}".format(bgarray=settings.bgarray, log=settings.log)
+    remove_run_file = True
+
+    temperror = settings.temperror
+    folder_dic = settings.folder_dic
+    log = settings.log
+
+    for run in to_be_transferred:
+        continue_rsync = True
+        folder = folder_dic[to_be_transferred[run]]
+        action = ("rsync -rahuL --stats {user}@{server}:{path}{run} {output}/ \
+            1>> {bgarray}/{log} 2>> {bgarray}/{errorlog} 2> {temperror}".format(
+                user=settings.user,
+                server=hpc_server,
+                path=folder["input"],
+                run=run,
+                output=folder["output"],
+                bgarray=settings.bgarray,
+                log=log,
+                errorlog=settings.errorlog,
+                temperror=temperror
+            )
+        )
+
+        missing = check_if_file_missing(folder, client, run)
+
+        if missing:
+            continue_rsync, remove_run_file = action_if_file_missing(folder, continue_rsync, remove_run_file, missing, run)
+
+        with open(bgarray_log_file, 'a') as log_file:
+            log_file.write("\n#########\nDate: {date}\nRun_folder: {run}\n".format(date=date, run=run))
+
+        if continue_rsync:
+            rsync_and_check(action, run, to_be_transferred[run], temperror, settings.wkdir, folder_dic, log)
+
+    return remove_run_file
+
+
+def delete_run_file(remove_run_file, run_file):
+    if remove_run_file:
+        os.system("rm {}".format(run_file))
```

Comment:
I don't think the if should be part of the function... And I don't think you even a need a function for this, since you can also use `os.remove()` (https://docs.python.org/3/library/os.html#os.remove): 
so in main:

```python
if remove_run_file:
   os.remove(run_file)

```

or just add the remove logic to `rsync_server_remote`